packages/patrol_log/lib/src/entry.dart

```diff
@@ -0,0 +1,24 @@
+abstract class Entry {
+  Entry({required this.timestamp, required this.type});
+
+  // ignore: avoid_unused_constructor_parameters
+  factory Entry.fromJson(Map<String, dynamic> json) {
+    throw UnimplementedError('fromJson is not implemented');
+  }
+
+  final DateTime timestamp;
+  final EntryType type;
+
+  Map<String, dynamic> toJson();
+
+  String pretty();
+
+  @override
+  String toString() => throw UnimplementedError('toString is not implemented');
+}
+
+enum EntryType {
+  step,
+  test,
+  log;
+}
```

Comment:
I guess that won't be necessary with sealed classes approach

Reply:
We need to somehow pass type of the step to json, and use this info in `parseEntry` in patrol_log_reader so this was my way. Do you have better idea?

Reply:
You're right, we still need that enum